pages/api/image.png.ts

```diff
@@ -26,19 +17,39 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
     height: 720,
     deviceScaleFactor: 1,
   });
+
   // Generate the full URL out of the given path (GET parameter)
-  await page.goto("https://google.com", {
+  await page.goto(req.body.url, {
     timeout: 15 * 1000,
   });
 
   // Generate image
   const data = await page.screenshot({
     type: "png",
   });
+
+  let domain = (new URL(req.body.url)).hostname;
+  domain = domain.replace(/^www\./, "");
+  domain = domain.split(".")[0];
+
+  const currentDate = new Date().toISOString().replace(/:/g, "-");
+
+  const fileName = `${domain}-${currentDate}.png`;
+
+  fs.writeFileSync(`public/images/${fileName}`, data);
```

Comment:
This won't work on the server - file systems aren't real in lambdas (all cleaned up post-request)

I was using the cache headers as a hack around this

Reply:
No prob. Will remove that.